lynx/src/main/java/com/github/pedrovgs/lynx/LynxConfig.java

```diff
@@ -55,10 +58,21 @@ public LynxConfig setMaxNumberOfTracesToShow(int maxNumberOfTracesToShow) {
   }
 
   public LynxConfig setFilter(String filter) {
+    if (filter == null) {
+      throw new NullPointerException("filter can't be null");
```

Comment:
Cool paragraph from Effective Java! To me, the problem with NPE is related with Java and Android developers. In Java if you see a NPE, first thing you think is: "fuck, I've a crash in my app! Lynx library has a bug..." I prefer to use IAE, but it's a cool argument :) I'll use NPE instead of IAE in my next library.